### ISSUE TITLE
BUG jquery included from the theme interferes with jquery included by the module

### ISSUE THREAD

**Issue by GuySartorelli** [dynamic/silverstripe-flexslider#210]
**Describe the bug**
If a theme includes its own jquery.js, there can be conflicts which result in the following error:

> Uncaught TypeError: jQuery(...).eq(...).flexslider is not a function

This seems to be caused by the jquery instance being used by the inline `Requirements::customScript()` in FlexSlider.php looking at the version of jquery injected by the theme, which is not the version which has had flexslider.js applied to it.

The ultimate result is that flexslider js doesn't apply to the carousel at all. In my case this means it is stuck on the first slide, and there are no visible controls.

**To Reproduce**
1. Include jquery.js in your theme (e.g. `<% require javascript('//code.jquery.com/jquery-3.3.1.min.js') %>`)
    - Probably this needs to be _after_ the `<% include FlexSlider %>` statement
3. Add flexslider per the documented instructions
4. Add some slides to a page and view it on the frontend
5. See error

**Expected behavior**
Flexslider should work without any console errors.

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04
 - Browser: Chromium, Firefox
 - Version [e.g. 22] _not sure what this is meant to be for? Version of browser? Of OS?_

**Workaround:**
Apply the following to your page controller:
```php
    protected function init()
    {
        // Don't include the jquery from this module's template
        Requirements::block('silverstripe/admin: thirdparty/jquery/jquery.js');
        // Ensure flexslider.js isn't executed until the theme's jquery.js has been applied
        Requirements::javascript('dynamic/flexslider: thirdparty/flexslider/jquery.flexslider-min.js', ['defer' => true]);
        parent::init();
    }
```

**Comment by muskie9** [comment]
@GuySartorelli sorry this fell through the cracks. I've noticed this repeatedly with some of our newer projects and similar to the work being done in the core modules to sure up client side dependency maintenance I'm looking at how we can best support the requirements of flexslider as it relates to the client dependencies that are packaged in Silverstripe. I think that's the most efficient way to ensure that a given module (flexslider) will work with a Silverstripe installation out of the box (assuming version constraints are enforced at the module level).

I'm curious if you think this is the way to go from a module standpoint so we can build a testing/QA methodology around requiring a consistent jQuery version for a community module knowing this isn't a CMS interface component, but rather something in the theme. I've hit similar things with the Comments module recently so this may be pretty common in the community.

**Comment by GuySartorelli** [comment]
> I'm looking at how we can best support the requirements of flexslider as it relates to the client dependencies that are packaged in Silverstripe. I think that's the most efficient way to ensure that a given module (flexslider) will work with a Silverstripe installation out of the box

The client dependencies in the silverstripe/admin module are _explicitly_ for the backend. They're not intended to be injected into the frontend. That's why things like `GridField` just don't work as front-end fields.

Since this is on the _front end_, imo it shouldn't depend on, require, or use any javascript bundled with Silverstripe at all. Silverstripe's javascript is all geared towards managing the admin section, with very few exceptions which are in specific modules.

It's not bad to provide jQuery either through a CDN or bundled with this module itself (not pulling from the admin module) _as an option_ - but it should absolutely be configurable so that if a theme already includes jQuery this module doesn't have to inject a _second_ jQuery out to the client. Basically the ideal approach IMO would be "I have this if you need it, but you can use your own if you'd prefer to"

**Comment by muskie9-opensource** [comment]
Good point. I know Userforms does something similar (I have a ticket to put in for that though). Client dependencies are always a bit finicky so I'll work through a configurable implementation that's a bit more developer friendly in our 4 release line.

I'm hoping to expand on our different slide types for a v5 and get that to stable sooner than alter and encapsulate these types of improvements as well.

**Comment by muskie9-opensource** [comment]
@GuySartorelli you have much more pressing things to do, however I have a draft in #212 and I'm generally curious if this is a "community acceptable approach" in your mind. In either case my tests work so I'll be merging and releasing, but with the forward thinking of CMS 5, this is something with UX type modules that could be worth informing from a community standpoint.

Robust UX depends on client side libraries, and as community developers we'll try to make things as "drop in ready" as we can, but there will always be exceptions. I'm curious how we can guide the community to spread their wings and explore/provide the best possible modules, but also make them adaptable to other developers' projects and their requirements.

**Comment by GuySartorelli** [comment]
Some of the wording in the docs is a little confusing - in particular I'd recommend saying "this is how you disable it" instead of "this is the default config" - developers will typically want to know what configuration _they_ can set rather than what has already been set. 

As a general approach though that PR is spot on, in my opinion. It works out of the box but allows people to use their own jQuery if the one from the module is causing grief in any manner.

**Comment by muskie9-opensource** [comment]
Thanks for the feedback, I'll update the docs (it's a great perspective and I agree). We'll get this reviewed and merged. I hope we can get this type of trend going in the community as I know this is something with some modules that can be a pain point.

**Comment by muskie9-opensource** [comment]
resolved via #212

**Comment by muskie9-opensource** [comment]
in release [4.2.0](https://github.com/dynamic/silverstripe-flexslider/releases/tag/4.2.0)
